src/Servant/Swagger/Internal.hs

```diff
@@ -146,70 +146,86 @@ addDefaultResponse400 pname = setResponseWith (\old _new -> alter400 old) 400 (r
+instance (ToSchema a, AllAccept cs, AllToResponseHeader hs, KnownNat c) => HasSwagger (Verb 'PATCH c cs (Headers hs a)) where
+  toSwagger = mkEndpoint "/" patch (fromIntegral (natVal (Proxy :: Proxy c)))
 
-instance AllAccept cs => HasSwagger (Patch cs ()) where
+instance AllAccept cs => HasSwagger (Verb 'PATCH c cs ()) where
   toSwagger = noContentEndpoint "/" patch
 
 -- -----------------------------------------------------------------------
 -- PUT
 -- -----------------------------------------------------------------------
 
-instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Put cs a) where
+instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Verb 'PUT c cs a) where
   toSwagger _ = toSwagger (Proxy :: Proxy (Put cs (Headers '[] a)))
 
-instance (ToSchema a, AllAccept cs, AllToResponseHeader hs) => HasSwagger (Put cs (Headers hs a)) where
-  toSwagger = mkEndpoint "/" put 200
+instance (ToSchema a, AllAccept cs, AllToResponseHeader hs, KnownNat c) => HasSwagger (Verb 'PUT c cs (Headers hs a)) where
+  toSwagger = mkEndpoint "/" put (fromIntegral (natVal (Proxy :: Proxy c)))
 
-instance AllAccept cs => HasSwagger (Put cs ()) where
+instance AllAccept cs => HasSwagger (Verb 'PUT c cs ()) where
   toSwagger = noContentEndpoint "/" put
 
 -- -----------------------------------------------------------------------
 -- POST
 -- -----------------------------------------------------------------------
 
-instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Post cs a) where
+instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Verb 'POST c cs a) where
   toSwagger _ = toSwagger (Proxy :: Proxy (Post cs (Headers '[] a)))
 
-instance (ToSchema a, AllAccept cs, AllToResponseHeader hs) => HasSwagger (Post cs (Headers hs a)) where
-  toSwagger = mkEndpoint "/" post 201
+instance (ToSchema a, AllAccept cs, AllToResponseHeader hs, KnownNat c) => HasSwagger (Verb 'POST c cs (Headers hs a)) where
+  toSwagger = mkEndpoint "/" post (fromIntegral (natVal (Proxy :: Proxy c)))
 
-instance AllAccept cs => HasSwagger (Post cs ()) where
+instance AllAccept cs => HasSwagger (Verb 'POST c cs ()) where
   toSwagger = noContentEndpoint "/" post
 
 instance (HasSwagger a, HasSwagger b) => HasSwagger (a :<|> b) where
   toSwagger _ = toSwagger (Proxy :: Proxy a) <> toSwagger (Proxy :: Proxy b)
 
+-- | Vault combinator does not change our specification at all.
+instance (HasSwagger sub) => HasSwagger (Vault :> sub) where
+  toSwagger _ = toSwagger (Proxy :: Proxy sub)
+
+-- | IsSecure combinator does not change our specification at all.
+instance (HasSwagger sub) => HasSwagger (IsSecure :> sub) where
+  toSwagger _ = toSwagger (Proxy :: Proxy sub)
+
+-- | RemoteHost combinator does not change our specification at all.
+instance (HasSwagger sub) => HasSwagger (RemoteHost :> sub) where
+  toSwagger _ = toSwagger (Proxy :: Proxy sub)
+
+-- | HttpVersion combinator does not change our specification at all.
```

Comment:
Can you change `HttpVersion` to `@'HttpVersion'@` for better haddock rendering?
Similarly for a few comment line above.